sidechain/consensus/common/src/is_descendent_of_builder.rs

```diff
@@ -0,0 +1,35 @@
+#[cfg(test)]
+use std::marker::PhantomData;
+
+#[cfg(test)]
+pub struct IsDescendentOfBuilder<Hash>(PhantomData<Hash>);
+#[cfg(test)]
+impl<'a, Hash: PartialEq> IsDescendentOfBuilder<Hash> {
+	#[cfg(test)]
+	/// Build the `is_descendent_of` closure for the fork-tree structure
+	/// to utilize when adding and removing nodes from the tree.
+	pub fn build_is_descendent_of(
+		_curr_block: Option<(&Hash, &Hash)>,
+	) -> impl Fn(&Hash, &Hash) -> Result<bool, ()> + 'a {
+		move |_base, _head| Ok(true)
+	}
+}
+
+#[cfg(test)]
+pub struct LowestCommonAncestorFinder<Hash>(PhantomData<Hash>);
+#[cfg(test)]
+impl<Hash: PartialEq + Default> LowestCommonAncestorFinder<Hash> {
+	#[cfg(test)]
+	/// Used by the `build_is_descendent_of` to find the LCA of two nodes in the fork-tree.
+	pub fn find_lowest_common_ancestor(_a: Hash, _b: Hash) -> Hash {
```

Comment:
Minor thing I forgot to address: First comes the doc string, and afterwards any proc_macro attribute. I wonder why this is not a lint yet. :) However, I will be lenient, as you will start using the stuff afterwards anyhow, and therefore be removing those flags. :)